### PR TITLE
fix: return error instead of panic for behaviors triggered by client

### PR DESCRIPTION
**Pull request by tkxkd0159**
## Description

closes: #XXXX

## Checklist:

- [x] I followed the [contributing guidelines](https://github.com/Finschia/finschia-sdk/blob/main/CONTRIBUTING.md) and [code of conduct](https://github.com/Finschia/finschia-sdk/blob/main/CODE_OF_CONDUCT.md).
- [x] I have added a relevant changelog to `CHANGELOG.md`
- [x] I have added tests to cover my changes.
- [ ] I have updated the documentation accordingly.
- [ ] I have updated API documentation `client/docs/swagger-ui/swagger.yaml`

### CODE REVIEW

File: x/fbridge/keeper/transfer.go [hunk 1]
```diff
@@ -23,7 +23,7 @@ func (k Keeper) handleBridgeTransfer(ctx sdk.Context, sender sdk.AccAddress, amo
 	}
 
 	if err := k.bankKeeper.SendCoinsFromAccountToModule(ctx, sender, types.ModuleName, token); err != nil {
-		panic(err)
+		return 0, err
 	}
 
 	if err := k.bankKeeper.BurnCoins(ctx, types.ModuleName, token); err != nil {
```
Comment (jaeseung-bae) [inline x/fbridge/keeper/transfer.go:29]: How about changing this case as well?

From:
```go
if err := k.bankKeeper.BurnCoins(ctx, types.ModuleName, token); err != nil {
    panic(fmt.Errorf("cannot burn coins after a successful send to a module account: %v", err))
```
To: 
> you can use %w to wrap another error
```go
if err := k.bankKeeper.BurnCoins(ctx, types.ModuleName, token); err != nil {
    return 0, fmt.Errorf("cannot burn coins after a successful send to a module account: %w", err)
```

**Review by 0Tech — COMMENTED** [review]
It could be nice to address other panics on x/fbridge. I found 2 points:

- critical
    - keeper/abci.go#L36: In theory, `updateRole()` can return error by registering two identical role suggestions. So we have to update the error handling or the chain halts.
- good-to-have
    - keeper/genesis.go#L22 and L28: `setRole()` and `setBridgeSwitch()` may return error because of its argument verification. It that case, `InitGenesis()` panics, instead of returning errors. However, if `InitGenesis()` returns any errors, eventually it will panic by the module. Hence, it would be good to return errors or modify the signature of `InitGenesis()` so it won't return any errors.

File: x/fbridge/keeper/transfer.go (hunk 1 above)
Comment (tkxkd0159) [reply x/fbridge/keeper/transfer.go:29]: this case must not be executed and cannot be triggered by client. So I think it's better to keep it.

File: x/fbridge/keeper/auth_test.go [hunk 2]
```diff
@@ -52,10 +52,6 @@ func TestAssignRole(t *testing.T) {
 	err = k.updateRole(ctx, types.RoleOperator, addrs[1])
 	require.NoError(t, err)
 	require.Equal(t, types.RoleMetadata{Guardian: 1, Operator: 1, Judge: 0}, k.GetRoleMetadata(ctx))
-
```
Comment (0Tech) [inline x/fbridge/keeper/auth_test.go:55]: Or, by confirming it does not return an error, we can recycle the obsolete test logic into a nice verification on the role update specification (i.e. it will silently ignore the suggestion, making it idempotent).

File: x/fbridge/keeper/auth.go [hunk 3]
```diff
@@ -61,10 +61,6 @@ func (k Keeper) addVote(ctx sdk.Context, proposalID uint64, voter sdk.AccAddress
 
 func (k Keeper) updateRole(ctx sdk.Context, role types.Role, addr sdk.AccAddress) error {
 	previousRole := k.GetRole(ctx, addr)
-	if previousRole == role {
```
Comment (0Tech) [inline x/fbridge/keeper/auth.go:64]: We may return at this point.

**Review by 0Tech — COMMENTED** [review]
And please mind keeper/genesis.go#L22.

File: x/fbridge/keeper/transfer.go (hunk 1 above)
Comment (jaeseung-bae) [reply x/fbridge/keeper/transfer.go:29]: Okay.

**Review by 0Tech — APPROVED** [review]

File: x/fbridge/keeper/abci.go [hunk 4]
```diff
@@ -32,11 +34,17 @@ func (k Keeper) EndBlocker(ctx sdk.Context) {
 		}
 
 		if types.CheckTrustLevelThreshold(k.GetRoleMetadata(ctx).Guardian, voteYes, guardianTrustLevel) || proposal.Proposer == k.GetAuthority() {
-			if err := k.updateRole(ctx, proposal.Role, sdk.MustAccAddressFromBech32(proposal.Target)); err != nil {
```
Comment (0Tech) [inline x/fbridge/keeper/abci.go:35]: Now, we can revert this change thanks to idempotent behavior of `updateRole()`.

**Review by 0Tech — APPROVED** [review]

**Review by jaeseung-bae — APPROVED** [review]
